storey/flow.py

```diff
@@ -1575,12 +1765,9 @@ def __init__(
             raise ValueError("ParallelExecution cannot be instantiated without at least one runnable")
 
         self.runnables = runnables
-        self._runnable_by_name = {}
-
-        self.max_processes = max_processes or os.cpu_count() or 16
-        self.max_threads = max_threads or 32
-
-        self._process_executor_by_runnable_name = {}
+        self.runnable_executor = RunnableExecutor(
+            max_processes=max_processes, max_threads=max_threads, pool_factor=pool_factor
+        )
```

Comment:
Actually, we need to move it to `_init`, because the graph is reusable. Otherwise, this will fail on graph termination and restart, which is what we do on rebalance.